src/04-test-class/index.test.ts

```diff
@@ -1,44 +1,70 @@
 // Uncomment the code below and write your tests
-// import { getBankAccount } from '.';
+import {
+  getBankAccount,
+  InsufficientFundsError,
+  TransferFailedError,
+  SynchronizationFailedError,
+} from '.';
 
 describe('BankAccount', () => {
   test('should create account with initial balance', () => {
-    // Write your test here
+    expect(getBankAccount(345).getBalance()).toBe(345);
   });
 
   test('should throw InsufficientFundsError error when withdrawing more than balance', () => {
-    // Write your test here
+    expect(() => getBankAccount(444).withdraw(445)).toThrow(
+      InsufficientFundsError,
+    );
   });
 
   test('should throw error when transferring more than balance', () => {
-    // Write your test here
+    expect(() =>
+      getBankAccount(444).transfer(449, getBankAccount(667)),
+    ).toThrow(InsufficientFundsError);
   });
 
   test('should throw error when transferring to the same account', () => {
-    // Write your test here
+    const account = getBankAccount(444);
+    expect(() => account.transfer(444, account)).toThrow(TransferFailedError);
   });
 
   test('should deposit money', () => {
-    // Write your test here
+    const account = getBankAccount(4);
+    expect(account.deposit(4).getBalance()).toBe(8);
   });
 
   test('should withdraw money', () => {
-    // Write your test here
+    const account = getBankAccount(5);
+    expect(account.withdraw(4).getBalance()).toBe(1);
   });
 
   test('should transfer money', () => {
-    // Write your test here
+    const account1 = getBankAccount(10);
+    const account2 = getBankAccount(0);
+    account1.transfer(5, account2);
+    expect(account2.getBalance()).toBe(5);
```

Comment:
`account1` balance should be verified after transfer as well:
```suggestion
    expect(account1.getBalance()).toBe(5);
    expect(account2.getBalance()).toBe(5);
```